include/powerloader/mirror.hpp

```diff
@@ -62,11 +62,29 @@ namespace powerloader
         }
     };
 
+    inline std::string strip_trailing_slash(const std::string& s)
+    {
+        if (s.size() > 0 && s.back() == '/' && s != "file://")
+        {
+            return s.substr(0, s.size() - 1);
+        }
+        return s;
+    }
+
     // mirrors should be dict -> urls mapping
-    struct POWERLOADER_API Mirror
+    class POWERLOADER_API Mirror
```

Comment:
This class looks to be used as an API for different implementations. However it does not have any virtual method, making me think that its usage would be more for factoring data members than for overriding behaviors. Therefore two questions:
- Is there a chance that we have other implementations in the future (different from HTTPMirror)?
- do we have to store different implementations in a same container?

Reply:
> Is there a chance that we have other implementations in the future (different from HTTPMirror)?

We already have OCIMorror and S3Mirror in the [`include/mirrors/*.hpp`](https://github.com/mamba-org/powerloader/tree/main/include/powerloader/mirrors) directory.

> do we have to store different implementations in a same container?

Currently all the kinds of mirror instances are stored in a `mirror_map_type` which is basically `std::map<string, std::vector<std::shared_ptr<Mirror>>>` so right now I would say yes.

There are alternative designs to how Mirror is handled, though we didnt try to improve that here for now.

Reply:
Ah indeed, I totally forgot these mirrors. Thus I agree, we can see how to improve this later.